### PR TITLE
Refactor handling for go-langserver installation

### PR DESCRIPTION
**Pull request by stamblerre**
Remove the code that handled this in the language server installation, and be careful to only install the tool specified by the user. Also, remove some unused import lines in the goLanguageServer.ts file.

Fixes microsoft/vscode-go#3028.

### CODE REVIEW

File: src/goLanguageServer.ts [hunk 1]
```diff
@@ -311,39 +298,42 @@ export function getLanguageServerToolPath(): string {
 		return;
 	}
 
-	// Get the path to gopls or any alternative that the user might have set for gopls.
-	const goplsBinaryPath = getBinPath('gopls');
-	if (path.isAbsolute(goplsBinaryPath)) {
-		return goplsBinaryPath;
-	}
```
Comment (ramya-rao-a) [inline src/goLanguageServer.ts]: There would still be merit in having the above. With the changes in this PR, if `goConfig['alternateTools']` is null or undefined, then we end up prompting the user to install `gopls` even if it was already installed

cc @stamblerre, @hyangah

File: src/goLanguageServer.ts [hunk 2]
```diff
@@ -311,39 +298,42 @@ export function getLanguageServerToolPath(): string {
...
 	if (goConfig['alternateTools']) {
 		const goplsAlternate = goConfig['alternateTools']['gopls'];
-		const golangserverAlternate = goConfig['alternateTools']['go-langserver'];
-		if (typeof goplsAlternate === 'string') {
+
+		// Check if the user has set the deprecated "go-langserver" setting.
+		if (goConfig['alternateTools']['go-langserver']) {
+			vscode.window.showErrorMessage(`The "go.alternateTools" setting for "go-langserver" has been deprecated.
+Please set "gopls" instead, and then reload the VS Code window.`);
+			return;
```
Comment (ramya-rao-a) [inline src/goLanguageServer.ts]: If the user had `gopls` installed in this case, then shouldn't we gracefully switch to that?
And also perhaps add a button to update the settings to remove `go-langserver` from the alternate tools setting?

File: src/goLanguageServer.ts [hunk 3]
```diff
@@ -311,39 +298,42 @@ export function getLanguageServerToolPath(): string {
...
+	const languageServerBinPath = getBinPath(languageServerOfChoice);
+	if (path.isAbsolute(languageServerBinPath)) {
+		return languageServerBinPath;
+	}
+
+	// Installation of gopls is supported. Other language servers must be installed manually.
+	if (languageServerOfChoice !== 'gopls') {
 		vscode.window.showErrorMessage(
-			`Cannot find the language server ${languageServerOfChoice}. Please install it and reload this VS Code window`
+			`Cannot find the language server ${languageServerOfChoice}. Please install it and reload this VS Code window.`
```
Comment (ramya-rao-a) [inline src/goLanguageServer.ts]: How about updating this to add `or use gopls` and add a button that will update the alternate tool settings and install `gopls` if it doesnt exist?

File: src/goLanguageServer.ts (hunk 2 above)
Comment (ramya-rao-a) [reply src/goLanguageServer.ts]: And if the user didn't had `gopls` installed, have a button to install `gopls`?

**Comment by ramya-rao-a** [comment]
The readme would need an update as well.

cc @stamblerre @hyangah

**Comment by ramya-rao-a** [comment]
Can we have the downstream change merged to this repo?

**Comment by ramya-rao-a** [comment]
Can we re-open this PR? The changes look good. I have added few suggested comments that can be handled in a separate PR

**Comment by stamblerre** [comment]
I contributed the PR to Go Nightly (https://github.com/golang/vscode-go/pull/2), but I can add those changes to that repository.

**Pull request closed without merging**